storage/remote/queue_manager_test.go

```diff
@@ -711,22 +711,26 @@ func TestShouldReshard(t *testing.T) {
 		startingShards                           int
 		samplesIn, samplesOut, lastSendTimestamp int64
 		expectedToReshard                        bool
+		sendDeadline                             model.Duration
 	}
 	cases := []testcase{
 		{
-			// Resharding shouldn't take place if the last successful send was > batch send deadline*2 seconds ago.
-			startingShards:    10,
+			// resharding shouldn't take place if we haven't successfully sent
+			// since the last shardUpdateDuration, even if the send deadline is very low
+			startingShards:    5,
 			samplesIn:         1000,
 			samplesOut:        10,
-			lastSendTimestamp: time.Now().Unix() - int64(3*time.Duration(config.DefaultQueueConfig.BatchSendDeadline)/time.Second),
+			lastSendTimestamp: time.Now().Unix() - int64(shardUpdateDuration),
```

Comment:
Was the `3*` chosen to guarantee the timestamp was before the calculated point?
In which case this should be `2*` or subtract a short interval.

Reply:
Yes, the here goal is to set the `lastSendTimestamp` value such that `shouldReshard` will return false. In the other test we want `shouldReshard` to be true. 

Luckily here calling `time.Now().Unix() - int64(shardupdate)` here and then later calling `minSendTimestamp := time.Now().Add(-1 * shardUpdateDuration).Unix()` in `shouldReshard` ensures that the value we've set for `lastSendTimestamp` is always older than that value in `shouldReshard`.